html/lesson4/tutorial.md

```diff
@@ -350,6 +350,232 @@ Add space around the images so there is some separation between them and the tex
+
+#### `flex-direction` property
+
+Flexbox is a system for laying out elements in one specific direction - either horizontally or vertically. The `flex-direction` property sets the direction of the main axis, defining the direction of the elements in the container.
+
+```css
+flex-direction: row | row-reverse | column | column-reverse;
+```
+
+- `row` - the axis starts on the left and ends on the right. This is the default value.
+- `row-reverse` - the axis starts on the right and ends on the left.
+- `column` - the main axis becomes vertical, starting at the top and ending at the bottom.
+- `column-reverse` - the main axis becomes vertical, starting at the bottom and ending at the top.
+
+  ![](assets/images/flex-direction.svg)
+
+By default, `flex-direction` is set to `row`, so the top menu items start lining up from left to right. The bottom menu is set to `row-reverse`, the main axis is reversed 180 degrees, and the items line up from right to left.
+
+#### `justify-content` property
+
+Controls the positioning of elements on the main axis, from `main-start` to `main-end`.
+
+```css
+justify-content: flex-start | flex-end | center | space-between | space-around |
+  space-evenly;
+```
+
+- `flex-start` - items are pressed to the start of the axis. This is the default value.
+- `flex-end` - elements are pressed to the end of the axis.
+- `center` - elements are centered on the axis.
+- `space-between` - elements are evenly distributed on the main axis. The first element is set to the beginning of the axis, and the last element to the end.
+- `space-around` - elements are distributed evenly, but the left and right elements lag behind the container borders by half the gap between the other elements.
+- `space-evenly` - the elements are distributed so that the distance between the elements, and from the outermost elements to the borders of the container, is the same
+
+  ![](assets/images/justify-content.jpeg)
+
+The most commonly used values, including the default value, are shown in the example. These include shifting content to the right (`flex-end`), centering (`center`), and evenly distributing space between elements (`space-between`).
+
+#### `align-items` property
+
+Controls the arrangement of elements along the transverse axis. This is the equivalent of `justify-content` for the main axis.
+
+```css
+align-items: stretch | flex-start | flex-end | center | baseline;
+```
+
+- `stretch` - elements are stretched to the full length of the axis. This is the default value.
+- `flex-start` - elements are pressed to the start of the axis.
+- `flex-end` - elements are pressed against the end of the axis.
+- `center` - elements are centered on the axis.
+- `baseline` - elements are aligned to the baseline of their text content.
+
+![](assets/images/align-items.svg)
```

Comment:
Many students have asked me about flex-box, so it might be beneficial. I do not think it is difficult for beginners. For creating a layout, it is essential to know.

Reply:
I think all properties nice to know, however, I added [flexboxfroggy](https://flexboxfroggy.com/) game we they can practice as well

Reply:
But I agree that we can simplify it; it is a modern and helpful approach. I mentioned that many students find it challenging to build layout styles; with the flex box, it will be much easier.

Reply:
I agree that it's not too difficult for beginners, that's not what I'm saying. It's more that if we try to include a complete explanation of everything in the introductory tutorials then each tutorial will be hundreds of pages long. One doesn't learn a language by reading a dictionary cover-to-cover, you know? I'd argue it's better to give just the information that they absolutely need, and then encourage them to explore the documentation later if interested.

Reply:
Ok, let me try to simplify it and then just add the link "Do you want to learn more? checkout this link with documentation "